src/ngx_http_lua_output.c

```diff
@@ -802,4 +802,99 @@ ngx_http_lua_flush_cleanup(void *data)
     ctx->flushing_coros--;
 }
 
+#ifndef NGX_LUA_NO_FFI_API
+//Pre-Condition: offset + len <= size of buf
+int
+ngx_http_lua_ffi_write(ngx_http_request_t *r, const char *buf, size_t offset,
+                       size_t len, char **err)
```

Comment:
Style: should use 4 space indentation here. This should be caught by the `ngx-releng` tool as well.